File: frontend/src/services/dashboardService.js
import api from '../api/axiosConfig';

export const dashboardService = {
  getDashboardData: async () => {
    const response = await api.get('/api/admin/dashboard');
    return response.data;
  },

  getReadingStatistics: async (period = '7d') => {
    const response = await api.get(`/api/admin/dashboard/reading-statistics?period=${period}`);
    return response.data;
  },
};

File: backend/src/main/java/com/storyapp/storyapp/service/DashboardService.java
package com.storyapp.storyapp.service;

import com.storyapp.storyapp.dto.response.AdminDashboardResponse;
import com.storyapp.storyapp.dto.response.ReadingStatisticResponse;

import java.util.List;

public interface DashboardService {

    AdminDashboardResponse getDashboardData();

    List<ReadingStatisticResponse> getReadingStatistics(String period);
}
